lib/IDE/ArgumentCompletion.cpp

```diff
@@ -95,14 +97,46 @@ bool ArgumentTypeCheckCompletionCallback::addPossibleParams(
 static bool hasParentCallLikeExpr(Expr *E, ConstraintSystem &CS) {
   E = CS.getParentExpr(E);
   while (E) {
-    if (E->getArgs() || isa<ParenExpr>(E) || isa<TupleExpr>(E) || isa<CollectionExpr>(E)) {
+    if (E->getArgs() || isa<ParenExpr>(E) || isa<TupleExpr>(E) ||
+        isa<CollectionExpr>(E)) {
       return true;
     }
     E = CS.getParentExpr(E);
   }
   return false;
 }
 
+/// The callee can be a double-applied function in the second apply (e.g.
+/// `f()(|)`). In that case, the normal callee locator will not be able to find
+/// a selected overload since an overload has been selected for the first apply
+/// but not the second. We try to find the function's declaration and type
+/// if it turns out to be a double apply.
+static std::optional<std::pair<ValueDecl *, AnyFunctionType *>>
+tryResolveDoubleAppliedFunction(CallExpr *OuterCall, const Solution &S) {
```

Comment:
`OuterCall` here can be null because the caller does a `dyn_cast`, you'll want to add an early return for that

Reply:
Shouldn't `ParentCall` always be a `CallExpr`? I should probably just convert `dyn_cast` into `cast`. WDYT?

Reply:
Hmm, in practice I believe so, I think I would still rather be defensive here and `dyn_cast` though